R/content.R

```diff
@@ -273,6 +273,56 @@ Content <- R6::R6Class(
       cat('content_item(client, guid = "', self$get_content()$guid, '")', "\n", sep = "")
       cat("\n")
       invisible(self)
+    },
+
+    #' @description Render or restart this content item.
+    refresh = function() {
+      if (is_rendered(self$content$app_mode)) {
+        self$render()
+      } else if (is_interactive(self$content$app_mode)) {
+        self$restart()
+      } else {
+        warning("Content app mode cannot be rendered or restarted.")
+      }
```

Comment:
It feels a little duplicative to have this else branch, but then also be checking within `render` and `restart` if they are of the correct type. 

I wonder if it would make sense to either collapse render and refresh here into this one method or to do `self$restart()` in the `else` here and live with the message being slightly off possibly?

I don't have a good answer, but seeing this + the stuff below makes it seem like we might be slightly off on this factoring. Maybe writing tests would help get this factored nicely?

Reply:
Yeah, I'm not sure. This is basically to handle the `static` app-mode, i.e. something that would neither be rendered or restarted. It feels weirdly duplicative with the other checks below, I agree.

I think the weirdness is… maybe largely a symptom of providing `refresh` that does both things and `render` / `restart` that each behave differently.

Reply:
I'll iterate on this more at the start of next week, to see if something else feels more ergonomically sensible.